src/utils/safe-store-set.js

```diff
@@ -0,0 +1,18 @@
+const logger = require('../common/logger')
+const store = require('../common/store')
+
+/**
+ * @template {unknown} R
+ * @param {string} key
+ * @param {unknown} value
+ * @param {() => Promise<R>|R|void} [onSuccessFn]
+ * @returns {Promise<R|void>}
+ */
+module.exports = async function (key, value, onSuccessFn) {
+  try {
+    store.set(key, value)
+    return onSuccessFn?.()
```

Comment:
I do think we should have some sort of success identifier, regardless of what it is. Part of the problem with the existing code is that we assume the values we're setting are actually being set even though they may not be. 

In general, I am not a fan of putting the burden on consumers for APIs. If we were to return a true/false, then the responsibility of control flow is on the consumer to check whether they set things successfully or not, and then call their own function. If we were going to do that, it would be better to just wrap all store.set() calls in a try/catch where it's used, instead of using safeStoreSet, but I feel like that's a lot for using a simple keyValue store.  

> return await is required here (because try-catch)

try-catch doesn't necessitate the use of `return await`. A `return await` keeps the awaited promise inside the call-stack, instead of throwing separately. 

By not awaiting the onSuccessFn, we require consumers to handle errors in their own function. We don't want to have safe-store-set be responsible for catching and logging errors for functions completely unrelated to the store, which is generally what the onSuccessFn are used for. e.g. set new apiUrl, then trigger ipfs-daemon restart.

Reply:
we *could* await the onSuccessFunction and then throw a separate error there while identifying the key used